src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -63,9 +60,10 @@ public DateTimeOffset GetLocalNow()
             }
 
             long localTicks = utcDateTime.Ticks + offset.Ticks;
-            if ((ulong)localTicks > (ulong)s_maxDateTicks)
+            const long MaxTicks = 3155378975999999999; // DateTime.MaxTicks
```

Comment:
As above, we don't need to inline the value in the source, this won't improve any performance. We should use `DateTime.MaxTicks` directly for better maintainability.

Reply:
This file is also used outside System.Private.CoreLib so it doesn't have access to internal fields/constants.

Reply:
I don't think it's worth propagating constants manually like this to make a tiny improvement in the corner cases where the codegen isn't perfect.